spec/controllers/url_redirects_controller_spec.rb

```diff
@@ -7,7 +7,7 @@
 
   before do
     @product = create(:product_with_pdf_file)
-    @url_redirect = create(:url_redirect, purchase: create(:purchase, email: "abCabC@abc.com", link: @product))
+    @url_redirect = create(:url_redirect, purchase: create(:test_purchase, email: "abCabC@abc.com", link: @product))
```

Comment:
shifted to a test purchase as it  avoids real charge validations and external payment logic that :purchase triggers.